### PR TITLE
KTOR-2468 Describe routing resolution strategy

### PR DESCRIPTION
**Pull request by rsinukov**

### CODE REVIEW

File: topics/resolution_algorithms.md [hunk 1]
```diff
@@ -1,42 +1,65 @@
...
-[Routing](Routing_in_Ktor.md) is organized in a tree with a recursive matching system that is capable of handling quite complex rules
-for request processing. The Tree is built with nodes and selectors. The Node contains handlers and interceptors,
-and the selector is attached to an arc which connects another node. If selector matches current routing evaluation context,
-the algorithm goes down to the node associated with that selector.
-
-Routing is built using a DSL in a nested manner:
+[Routing](Routing_in_Ktor.md) is organized in a tree with a recursive matching system that is capable of handling quite
+complex rules for request processing. The Tree is built with nodes that contain selector and optional handler. Selectors
+are used to pick a route based on request. Every selector has quality, usually from 0.0 to 1.0, with one special case
+of "transparent quality" (-1.0). This quality is useful when you need to wrap your subtree in a block, but this block
```
Comment (hhariri) [inline topics/resolution_algorithms.md:6]: This part would need clarifying. It introduces the concept of "block" but not clear what a block is. Also, doesn't really explain what the values of 0 to 1 mean.

File: topics/resolution_algorithms.md [hunk 2]
```diff
@@ -1,42 +1,65 @@
...
+should not change routes priority. Example of it can be `route("/")` block in the middle of a routing tree.
 
+Example tree:
```
Comment (hhariri) [inline topics/resolution_algorithms.md:9]: I think examples need to be built up without any nesting, as a first step. Once the aspect of quality is clear, then introduce nesting/blocks

**Review by hhariri — CHANGES_REQUESTED** [review]
I've not done any kind of proofreading, just focusing right now on the explanations.

File: topics/resolution_algorithms.md (hunk 1 above)
Comment (rsinukov) [reply topics/resolution_algorithms.md]: Updated it to "routing block". I believe the next sentence gives a good example.

File: topics/resolution_algorithms.md (hunk 2 above)
Comment (rsinukov) [reply topics/resolution_algorithms.md]: Added a simple example to show how quality is used.

**Review by hhariri — APPROVED** [review]

**Pull request merged**